src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs

```diff
@@ -171,15 +181,103 @@ public override async Task ExecuteAsync(InstanceDetailsViewModel model)
                 }
             }
 
+            if (!instance.VersionHasServiceControlAuditFeatures)
+            {
+                auditViewModel = auditUpgradeViewModelFactory(instance.Name);
+                if (windowManager.ShowInnerDialog(auditViewModel) == true)
+                {
+                    upgradeOptions.InstallNewAuditSidecar = true;
```

Comment:
Does this need to be added to the upgrade options? It only ever gets referred to from this method

Reply:
Possibly not. I wasn't sure if it is needed for other purposes. I can remove for now.